### PR TITLE
Bugfix - Some levels never end if the language is non-english.

### PR DESCRIPTION
**Pull request by daynew**
Bug: When a non-English user would finish some [levels](https://studio.code.org/s/coursee-2017/stage/20/puzzle/5), the pop-up which asks them to move on to the next level never appears. 
Cause: Around 50 levels use the `Success condition` field to define custom Javascript for detecting when a level is complete. These levels were waiting for `Studio.victoryText == "You lose!"` to be true, but this never occurs for non-English users because this string is translated.
Fix: Update the scripts to depend on boolean flags instead which indicate whether the user has won or lost.
* Updates levels to use `Studio.trackedBehavior.hasWonGame` and `Studio.trackedBehavior.hasLostGame`

## Links
- [jira](https://codedotorg.atlassian.net/browse/FND-1166)

## Testing story
* Manually tested.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/studio/studio.js [hunk 1]
```diff
@@ -5382,8 +5382,17 @@ Studio.setScoreText = function(opts) {
...
+ * Sets up Studio.victoryText and Studio.victoryValue
+ * Studio.victoryText is the text to display to a user when their level has met it's win or lose condition
+ * Studio.victoryValue is either 'win' or 'lose' despite the language settings for the page. This can be used to
+ *  determine if an end condition has been reached.
+ * @param {string} - text The translated string to display to the user when their game has reached an end condition.
+ * @param {string} - value Either 'win' or 'lose' depending on the end condition which was reached.
+ */
+Studio.setVictoryText = function(text, value) {
+  Studio.victoryText = text;
+  Studio.victoryValue = value;
```
Comment (Hamms) [inline apps/src/studio/studio.js]: how feasible would it be for `value` to be a boolean, instead? Something like `Studio.isVictory`?

Comment (daynew) [reply apps/src/studio/studio.js]: Thanks for asking about this. I investigated further and found some boolean flags which already exist. Only the levels need to be updated in this change.

**Review by Hamms — APPROVED** [review]

**Review by islemaster — APPROVED** [review]
:tada: Awesome, it's great that you found an existing feature we can depend on here.

I wonder what we can do to prevent levelbuilders from leaning on `victoryText` in the future - is this a training issue, or can we remove `victoryText` as a global so it's no longer available to them? (Either way I think this is a followup discussion, not blocking this PR.)

**Comment by daynew** [comment]
> 🎉 Awesome, it's great that you found an existing feature we can depend on here.
> 
> I wonder what we can do to prevent levelbuilders from leaning on `victoryText` in the future - is this a training issue, or can we remove `victoryText` as a global so it's no longer available to them? (Either way I think this is a followup discussion, not blocking this PR.)

Yeah, Elijah mentioned to me that we don't really provide an API for our level builders, so that might be an area worth exploring.

**Pull request merged**